examples/C/ssids_topology.c

```diff
@@ -0,0 +1,80 @@
+/* examples/C/ssids_topology.c - Example code for SPRAL_SSIDS package */
+#include "spral.h"
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+
+int main(void) {
+   /* Derived types */
+   void *akeep, *fkeep;
+   struct spral_ssids_options options;
+   struct spral_ssids_inform inform;
+
+   /* Initialize derived types */
+   akeep = NULL; fkeep = NULL; /* Important that these are NULL to start with */
+   spral_ssids_default_options(&options);
+   options.array_base = 1; /* Need to set to 1 if using Fortran 1-based indexing */
+
+   /* Data for matrix:
+    * ( 2  1         )
+    * ( 1  4  1    1 )
+    * (    1  3  2   )
+    * (       2 -1   )
+    * (    1       2 ) */
+   bool posdef = false;
+   int n = 5;
+   int64_t ptr[]   = { 1,        3,             6,         8,   9,  10 };
+   int row[]    = { 1,   2,   2,   3,   5,   3,   4,    4,   5   };
+   double val[] = { 2.0, 1.0, 4.0, 1.0, 1.0, 3.0, 2.0, -1.0, 2.0 };
+
+   /* The right-hand side with solution (1.0, 2.0, 3.0, 4.0, 5.0) */
+   double x[] = { 4.0, 17.0, 19.0, 2.0, 12.0 };
+
+   /* Fake topology */
+   #define nregions 2
+   struct spral_numa_region regions[nregions];
+   regions[0].nproc = 1;
+   regions[0].ngpu = 0;
+   regions[0].gpus = NULL;
+   regions[1].nproc = 1;
+   regions[1].ngpu = 2;
+   int gpus[] = {0,1};
+   regions[1].gpus = (int *) &gpus;
```

Comment:
A bit surprised about this, is this an explicit way to do the usual pointer decay that you get with just `regions[1].gpus = gpus`?

Reply:
Sorry being sloppy for no reason